README.md

```diff
@@ -141,15 +142,130 @@ for (index, game) in myFavoriteZeldaGames.enumerate() {
+
+![Array Type Inference](array-type-inference.png)
+
+## Loops and Calculations
+
+You typically use loops to undertake more complex repetitive processing than printing the value of a variable, such as calculations.
+
+Say you have a list of numbers and want to know what the square of each one and add that value to a new list. You create one array containing the original list of numbers and a placeholder array for the list of squared numbers. Then iterate through the array containing the list of numbers, perform the necessary calculations to find it's square and add that value to the second array. Let's look at some code!
+
+```swift
+let numbers = [1, 2, 3, 4, 5]
+var squaredNumbers: [Int] = []
+
+for number in numbers {
+    let squaredNumber = number * number
+    squaredNumbers.append(squaredNumber)
+}
+
+print(squaredNumbers)
+// prints "[1, 4, 9, 16, 25]"
+```
+
+For the first `numbers` array, you use the `let` keyword as you don't want to change the values in the array. As you don't know what the values of the second `squaredNumbers` array will be yet, you initialize it first with no values. Inside the `for in` loop you generate the square of the value at the current index of the array and assign it to another unchanging variable. The next line introduces something called a _method_, which you can think of as a pre-written convenient way of performing actions on something. In this particular case `.append` adds a new value to the end of the `squaredNumbers` array.
+
+This last example introduced a couple of new concepts, if you want to understand better what a line of code is doing then feel free to add `print` and `if` statements to inspect values (this is called 'debugging'). For example to see the value of `squaredNumber` you could add `print (squaredNumber)` inside the loop.
+
+Maybe one of your friends deserves a special mention for being so awesome, so you need to find them in the `friends` list, and when you do, print a message for them:
+
+```swift
+let friends = ["Chris", "Billy", "Jessica"]
+
+for friend in friends {
+    print("We are inside the for loop, friend: \(friend)")
+
+    if friend == "Billy" {
+        print("Hello \(friend), you are awesome!")
+    }
+}
+
+// We are inside the for loop, friend: Chris
+// We are inside the for loop, friend: Billy
+// Hello Billy, you are awesome!
+// We are inside the for loop, friend: Jessica
+```
+
+In the above, you iterate the `friends` list printing the value of each index in the array. If the current value is equal to "Billy", then inside an if statement you print them a special greeting.
+
+## Loops and Functions
+
+In unit 1, lesson 8 you learned about functions, a convenient way to package functionality into reusable components. You can add any code can into a function, including loops.
+
```

Comment:
Need to remove one "can" here. I can remove this myself as well, making note here so I don't forget.

Taking out unit 1, lesson 8 to replace with the following is also something I can do (or you if you get to it first).
[prior lesson](https://github.com/learn-co-curriculum/swift-functions-readme)